forge/src/executor/inspector/debugger.rs

```diff
@@ -52,21 +52,19 @@ impl Debugger {
     /// Builds the instruction counter map for the given bytecode.
     // TODO: Some of the same logic is performed in REVM, but then later discarded. We should
     // investigate if we can reuse it
-    pub fn build_ic_map(&mut self, address: &Address, code: &Bytes) {
+    pub fn build_ic_map(&mut self, spec: SpecId, address: &Address, code: &Bytes) {
+        let opcode_infos = spec_opcode_gas(spec);
         let mut ic_map: BTreeMap<usize, usize> = BTreeMap::new();
 
         let mut i = 0;
         let mut cumulative_push_size = 0;
         while i < code.len() {
             let op = code[i];
             ic_map.insert(i, i - cumulative_push_size);
-            match OpCode::is_push(op) {
-                Some(push_size) => {
-                    // Skip the push bytes
-                    i += push_size as usize;
-                    cumulative_push_size += push_size as usize;
-                }
-                None => (),
+            if opcode_infos[op as usize].is_push {
```

Comment:
let's add a note linking to how revm does it, in case we forget? https://github.com/bluealloy/revm/blob/007b8807b5ad7705d3cacce4d92b89d880a83301/crates/revm/src/interpreter/contract.rs#L114-L115